pkg/controllers/workapplier/availability_tracker.go

```diff
@@ -247,6 +254,41 @@ func trackPDBAvailability(curObj *unstructured.Unstructured) (ManifestProcessing
 	return ManifestProcessingAvailabilityResultTypeNotYetAvailable, nil
 }
 
+// trackServiceExportAvailability tracks the availability of a service export in the member cluster.
+// It is available if the ServiceExportValid condition is true (will be false if annotation value is invalid).
+// If the weight is not 0, ServiceExportValid condition must be true and the ServiceExportConflict condition must be false.
+func trackServiceExportAvailability(curObj *unstructured.Unstructured) (ManifestProcessingAvailabilityResultType, error) {
+	var svcExport fleetnetworkingv1alpha1.ServiceExport
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(curObj.Object, &svcExport); err != nil {
+		return ManifestProcessingAvailabilityResultTypeFailed, controller.NewUnexpectedBehaviorError(err)
+	}
+
+	// Check if ServiceExport is valid and up to date
+	svcExportObj := klog.KObj(curObj)
+	validCond := meta.FindStatusCondition(svcExport.Status.Conditions, string(fleetnetworkingv1alpha1.ServiceExportValid))
+	if !condition.IsConditionStatusTrue(validCond, svcExport.Generation) {
+		klog.V(2).InfoS("Still need to wait for ServiceExport to be valid", "serviceExport", svcExportObj, "validCondition", validCond)
+		return ManifestProcessingAvailabilityResultTypeNotYetAvailable, nil
+	}
+
+	weight, err := objectmeta.ExtractWeightFromServiceExport(&svcExport)
+	if err != nil {
+		klog.V(2).InfoS("ServiceExport has invalid weight", "serviceExport", svcExportObj, "error", err)
+		return ManifestProcessingAvailabilityResultTypeFailed, err
```

Comment:
```suggestion
		return ManifestProcessingAvailabilityResultTypeNotYetAvailable, err
```
ManifestProcessingAvailabilityResultTypeFailed means the checking the availability itself is failed.